app/models/solidus/gateway/braintree_gateway.rb

```diff
@@ -190,6 +196,49 @@ def cancel(response)
     end
 
     private
+
+    def add_payment_method(payment)
+      source = payment.source
+
+      user = payment.order.user
+      email = user ? user.email : payment.order.email
+      address = (payment.source.address || payment.order.bill_address).try(:active_merchant_hash)
+
+      result = braintree_gateway.payment_method.create(
+        payment_method_nonce: payment.payment_method_nonce,
+        customer_id: user.braintree_customer_id,
```

Comment:
`user` could be `nil` here, right?

Reply:
The method is only called when the `braintree_customer_id` is present. See line `73`